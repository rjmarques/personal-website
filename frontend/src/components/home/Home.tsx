import { Button } from "antd";
import { MailOutlined } from "@ant-design/icons";
import React from "react";

import "./Home.less";

interface IProps {
  goToContact: () => void;
}

const IMG_CLASS = "lazy";

document.addEventListener("DOMContentLoaded", () => {
  const lazyImage = document.querySelectorAll(`img.${IMG_CLASS}`)[0];

  const lazyImageObserver = new IntersectionObserver((entries) => {
    if (entries.length > 1) {
      throw new Error("There can be only one!");
    }

    const entry = entries[0];
    if (entry.isIntersecting) {
      const img = entry.target as HTMLImageElement;
      img.src = img.dataset.src || "";
      img.srcset = img.dataset.srcset || "";
      img.classList.remove(IMG_CLASS);
      lazyImageObserver.unobserve(img);
    }
  });

  lazyImageObserver.observe(lazyImage);
});

const Home = (props: IProps) => (
  <section className="home">
    <div className="content">
      <div className="home-content">
        <div className="home-hero-wrapper">
          <img
            id="home-hero-image"
            className={IMG_CLASS}
            alt="ric"
            src={require("./img/me_placeholder.png")}
            data-src={require("../../assets/img/me.jpg")}
          />
        </div>
        <div className="home-description materialize">
          <h1>Ricardo Marques</h1>
          <span className="home-sub-title">Full Stack Developer</span>
          <p>
            I specialize in Objected Oriented Development and Web Development. I
            stand for quality and love spending time optimizing every little
            piece of code.
          </p>
          <div className="home-buttons">
            <Button
              type="primary"
              icon={<MailOutlined />}
              size="large"
              onClick={props.goToContact}
            >
              Contact me
            </Button>
          </div>
        </div>
      </div>
    </div>
  </section>
);

export default Home;
