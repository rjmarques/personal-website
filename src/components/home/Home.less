@import "../../theme.less";

.Home {
  background: url(./img/banner.png) no-repeat scroll 0 0 / cover;
  min-height: 640px;
  position: relative;

  &::before {
    content: "";
    background-color: @main-blue;
    height: 100%;
    opacity: 0.7;
    position: absolute;
    width: 100%;
    z-index: 1;
  }

  p {
    font-size: 16px;
    line-height: 24px;
    letter-spacing: 0.1px;
    margin: 40px 0px;
  }

  .Home-content {
    background-color: #fff;
    border-top: 9px solid @primary-color;
    box-shadow: 0 2px 30px 5px rgba(51, 51, 51, 0.38);
    display: inline-block;
    margin-top: 150px;
    padding: 46px 90px 0 67px;
    position: relative;
    vertical-align: bottom;
    z-index: 10;
  }

  .Home-mugshot {
    // only when collums collapse -> border-bottom: 1px solid @light-grey;
    height: 445px;
    object-fit: none;
    width: 100%;
  }

  .Home-sub-title {
    color: @primary-color;
    display: block;
    font-size: 16px;
    margin: 8px 0px 0px 2px;
  }

  .Home-description {
    padding: 50px 60px 0px 60px;
  }

  .Home-buttons {
    > * {
      margin-right: 25px;
    }
  }
}
