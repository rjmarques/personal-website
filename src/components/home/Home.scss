@use "../../theme.scss" as *;
@use "sass:color";

.home {
  position: relative;

  // Background image for larger screens
  @include for-size(laptop-small-up) {
    background: url('./img/banner.png') no-repeat scroll 0 0 / cover;
  }

  &::before {
    content: "";
    background-color: $main-blue;
    height: 100%;
    opacity: 0.85;
    position: absolute;
    width: 100%;
    z-index: 1;

    @include for-size(laptop-small-up) {
      opacity: 0.7;
    }
  }

  p {
    line-height: 24px;
    letter-spacing: 0.1px;
    margin: 20px 0px 40px 0px;

    @include for-size(desktop-up) {
      font-size: 16px;
      margin: 40px 0px;
    }
  }

  .home-content {
    background-color: $white;
    border-top: 9px solid $primary-color;
    box-shadow: 0 2px 30px 5px rgba(51, 51, 51, 0.38);
    display: inline-grid;
    position: relative;
    vertical-align: bottom;
    z-index: 10;

    grid-template-columns: 1fr;
    margin-top: 110px;
    text-align: center;
    padding: 30px;

    @include for-size(laptop-small-up) {
      padding: 50px 80px;
    }

    @include for-size(laptop-up) {
      grid-template-columns: 1fr 2fr;
      text-align: left;
      padding: 46px 70px 0 67px;
    }

    @include for-size(desktop-up) {
      margin-top: 150px;
      padding: 46px 90px 0 67px;
    }
  }

  .home-hero-wrapper {
    border-bottom: 1px solid $light-grey;

    @include for-size(laptop-up) {
      border: none;
    }
  }

  #home-hero-image {
    height: auto;
    width: 200px;

    @include for-size(tablet-up) {
      width: 240px;
    }

    @include for-size(desktop-up) {
      width: 300px;
    }
  }

  .home-sub-title {
    color: $primary-color;
    display: block;
    font-size: 16px;
    font-weight: 600;
    margin: 8px 0px 0px 2px;
  }

  .home-description {
    padding-top: 20px;

    @include for-size(laptop-up) {
      padding: 50px 20px 0px 40px;
    }

    @include for-size(desktop-up) {
      padding: 50px 40px 0px 60px;
    }
  }

  .home-buttons {
    > * {
      font-weight: 600;

      @include for-size(tablet-up) {
        margin-left: 13px;
        margin-right: 13px;
        margin-bottom: 0px;
      }

      @include for-size(laptop-up) {
        margin-left: 0px;
        margin-right: 25px;
      }
    }
  }
}
