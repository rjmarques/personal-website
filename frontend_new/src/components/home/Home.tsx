"use client";

import React from "react";
import Image from "next/image";
import { Button } from "@mui/material";
import { Mail } from "@mui/icons-material";
import meImage from "../../assets/img/me.jpeg";
import "./Home.scss";

interface IProps {
  goToContact: () => void;
}

const Home = (props: IProps) => {

  return (
    <section className="home">
      <div className="content">
        <div className="home-content">
          <div className="home-hero-wrapper">
            <Image
              id="home-hero-image"
              alt="ric"
              src={meImage}
              priority
            />
          </div>
          <div className="home-description materialize">
            <h1>Ricardo Marques</h1>
            <span className="home-sub-title">Full Stack Developer</span>
            <p>
              I specialize in Objected Oriented Development and Web Development.
              I stand for quality and love spending time optimizing every little
              piece of code.
            </p>
            <div className="home-buttons">
              <Button
                variant="contained"
                startIcon={<Mail />}
                size="large"
                onClick={props.goToContact}
              >
                Contact me
              </Button>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Home;
