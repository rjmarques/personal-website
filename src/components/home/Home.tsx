import { Button, Col, Row } from "antd";
import * as React from "react";

import "./Home.less";

class Home extends React.Component {
  public render() {
    return (
      <section className="Home">
        <div className="content">
          <div className="Home-content">
            <Row>
              <Col xs={24} sm={24} md={8} lg={8} xl={8}>
                <img className="Home-mugshot" src={require("./img/me.jpg")} />
              </Col>
              <Col xs={24} sm={24} md={16} lg={16} xl={16}>
                <div className="Home-description">
                  <h1>Ricardo Marques</h1>
                  <span className="Home-sub-title">Full Stack Developer</span>
                  <p>
                    I specialize in Objected Oriented Development and Web
                    Development. I stand for quality and love spending time
                    optimizing every little piece of code.
                  </p>
                  <div className="Home-buttons">
                    <Button type="primary" size="large">
                      Contact me
                    </Button>
                    <Button icon="download" size="large">
                      Download CV
                    </Button>
                  </div>
                </div>
              </Col>
            </Row>
          </div>
        </div>
      </section>
    );
  }
}

export default Home;
