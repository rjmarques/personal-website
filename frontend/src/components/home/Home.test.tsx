import React from "react";
import renderer from "react-test-renderer";

import Home from "./Home";

describe("renders without crashing", () => {
  it("renders correctly", async () => {
    const goToContact = jest.fn();
    const component = renderer.create(<Home goToContact={goToContact} />);
    expect(component).toMatchSnapshot();
  });
});
