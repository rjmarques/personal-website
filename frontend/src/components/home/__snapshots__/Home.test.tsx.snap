// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders correctly 1`] = `
<section
  className="home"
>
  <div
    className="content"
  >
    <div
      className="home-content"
    >
      <div
        className="home-hero-wrapper"
      >
        <img
          alt="ric"
          className="lazy"
          data-src="me.jpg"
          id="home-hero-image"
          src="me_placeholder.png"
        />
      </div>
      <div
        className="home-description materialize"
      >
        <h1>
          Ricardo Marques
        </h1>
        <span
          className="home-sub-title"
        >
          Full Stack Developer
        </span>
        <p>
          I specialize in Objected Oriented Development and Web Development. I stand for quality and love spending time optimizing every little piece of code.
        </p>
        <div
          className="home-buttons"
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-lg"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-label="mail"
              className="anticon anticon-mail"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="mail"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                />
              </svg>
            </span>
            <span>
              Contact me
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
